src/middlewared/middlewared/plugins/datastore/read.py

```diff
@@ -230,18 +230,15 @@ def _get_queryset_joins(self, table):
             else:
                 data = await self.middleware.call(extend, data)
 
-        return data
+        return data if not select else {k: v for k, v in data.items() if k in select}
 
     def _strip_prefix(self, k, field_prefix):
         return k[len(field_prefix):] if field_prefix and k.startswith(field_prefix) else k
 
-    def _serialize_row(self, obj, table, aliases, select):
+    def _serialize_row(self, obj, table, aliases):
         data = {}
 
         for column in table.c:
-            if select and column.name not in select:
```

Comment:
We should not remove this as this will skip handling relationships if we are not selecting them